persistence-commons/persistence-commons-postgresql/src/main/java/org/kie/kogito/persistence/postgresql/PostgresStorage.java

```diff
@@ -73,7 +75,7 @@ public PostgresStorage(String cacheName, CacheEntityRepository repository, Objec
 
     @Override
     public Query<V> query() {
-        throw new UnsupportedOperationException("Custom query not available in PostgresSQL");
+        return query;
```

Comment:
I think this should return always a new instance of `PostgresQuery` and `query` field should be removed
```suggestion
        return new PostgresQuery<>(cacheName, repository, mapper, type);
```

Reply:
+1

Reply:
Good catch... it'd be fun trying to work out why filter/sorting was stuffed on re-use.